prow/test/integration-mkpj.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script loads Prow configuration YAML and validates that the
+# appropriate ProwJob is created from that configuration.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# We are told where the `mkpj` binary lives by Bazel
+mkpj=$1
+# We can either validate current behavior or update
+# the recorded files
+action=$2
+if [[ "${action}" != "VALIDATE" && "${action}" != "GENERATE" ]]; then
+    echo "[ERROR] Action must be either \`VALIDATE\` or \`GENERATE\`, not ${action}"
+    exit 1
+fi
+
+output="$( mktemp -d )"
+
+function cleanup() {
+    returnCode="$?"
+    rm -rf "${output}" || true
+    exit "${returnCode}"
+}
+trap cleanup EXIT
+
+function validate() {
```

Comment:
Consider moving this into a go program?

Reply:
I considered it, but it seemed overkill for right now. If this needs to grow, we can think about it?